src/Microsoft.TemplateEngine.Core/Util/ProcessorState.cs

```diff
@@ -53,14 +51,12 @@ public ProcessorState(Stream source, Stream target, int bufferSize, int flushThr
             CurrentBuffer = new byte[bufferSize];
             CurrentBufferLength = source.Read(CurrentBuffer, 0, CurrentBuffer.Length);
 
-            Encoding encoding = EncodingUtil.Detect(CurrentBuffer, CurrentBufferLength, out _bom);
-            Encoding = encoding;
-            CurrentBufferPosition = _bom.Length;
-            CurrentSequenceNumber = _bom.Length;
-            if (_bom.Length > 0)
-            {
-                _isBomNeeded = true;
-            }
+            Encoding encoding = EncodingUtil.Detect(CurrentBuffer, CurrentBufferLength, out byte[] bom);
+            EncodingConfig = new EncodingConfig(Config, encoding);
```

Comment:
I would consider a design pattern that would abstracts away from callers the dealing with `stream` interactions.  At a minimum, such overloads, as done in standard I\O .NET types.